src/java/org/apache/cassandra/db/lifecycle/Tracker.java

```diff
@@ -604,6 +604,15 @@ private void notify(INotification notification)
             subscriber.handleNotification(notification, this);
     }
 
+    public void publishMetrics(long bytesInserted, long partitionsRead, double flushSize, double sstablePartitionReadLatencyNanos, double flushTimePerKbNanos)
```

Comment:
It appears unnecessary for the tracker to understand what the metrics are. In fact, most likely the CFS shouldn't choose what to post either.

How about letting the CFS's `metric` object create the `MetricsNotification` instance, and have the CFS call that and pass it on to the tracker to `notify` (if you want to avoid changing its visibility, you could still add `publishMetrics` that accepts `MetricsNotification` as a parameter and defers to `notify`)?

Reply:
`ColumnFamilyStore.publishMetrics()` now calls `metrics().createMetricsNotification()` and passes the `MetricsNotification` to `Tracker.publishMetrics()`.